include/deal.II/lac/vector_memory.h

```diff
@@ -63,10 +65,38 @@ DEAL_II_NAMESPACE_OPEN
  * only marks them as unused and allows them to be reused next time a vector
  * is requested.
  *
- * Yet other classes, when implemented, could provide even other strategies
- * for memory management.
  *
- * @author Guido Kanschat, 1998-2003
+ * <h3> Practical use </h3>
+ *
+ * Classes derived from this base class return pointers to new vectors
+ * via the VectorMemory::alloc() function, and re-claim the vector
+ * when it is returned via VectorMemory::free(). These two functions
+ * therefore play a similar role as @p new and @p delete. This includes
+ * the usual drawbacks: It is simple to forget to call VectorMemory::free()
+ * at the end of a function that uses this facility, or to forget it in an
+ * @p if branch of the function where one has an early @p return from
+ * the function. In both cases, a memory leak results: a correct piece of
```

Comment:
I'm not a native speaker, but "a memory leak results" sounds weird to me. Wouldn't it be "a memory leak occurs"?

Reply:
I think that "occurs" sounds a _bit_ more natural in this context. When using "results" I would likely have written it as "In both cases, this results / would result in a memory leak"

Reply:
The latter two would also be the choices I would have made with "results", thanks for the confirmation @jppelteret ;-)